File: backend/src/main/kotlin/com/moa/backend/model/dto/CommentDto.kt
package com.moa.backend.model.dto

import com.moa.backend.model.slim.IdeaSlimDto
import com.moa.backend.model.slim.UserSlimDto
import java.util.*

data class CommentDto (

    var id: Long = 0,

    var creationDate: Date,

    var text: String,

    var owner: UserSlimDto,

    var idea: IdeaSlimDto,

    var likes: MutableList<UserSlimDto>?,
)

File: frontend/src/app/models/dto/commentDto.ts
import { IdeaSlimDto } from '../slimDto/ideaSlimDto';
import { UserSlimDto } from '../slimDto/userSlimDto';

export interface CommentDto {
  id?: number;
  creationDate?: Date;
  text?: string;
  owner?: UserSlimDto;
  idea?: IdeaSlimDto;
  likes?: UserSlimDto[];
  isEdited?: boolean;
}
